bubble_analyser/default.py

```diff
@@ -0,0 +1,161 @@
+import tomllib
+from pprint import pprint
+
+import matplotlib.pyplot as plt
+import numpy as np
+from scipy import ndimage
+from skimage import (
+    color,
+    filters,
+    io,
+    measure,
+    morphology,
+    segmentation,
+    transform,
+)
+from skimage.io import imsave
+
+
+def extendedmin(img, H):
+    mask = img.copy()
+    marker = mask + H
+    hmin = morphology.reconstruction(marker, mask, method="erosion")
+    return morphology.local_minima(hmin)
+
+
+def imposemin(img, minima):
+    marker = np.full(img.shape, np.inf)
+    marker[minima == 1] = 0
+    mask = np.minimum((img + 1), marker)
+    return morphology.reconstruction(marker, mask, method="erosion")
+
+
+def default(image_path, params):
+    # Read the input image
+    img = io.imread(image_path)
+
+    # Extract parameters from the params dictionary
+    se = morphology.disk(
+        params["Morphological_element_size"]
+    )  # Structuring element for morphological operations
+    nb = params["Connectivity"]  # Neighborhood connectivity (4 or 8)
+    marker_size = params["Marker_size"]  # Marker size for watershed segmentation
+    px2mm = params["px2mm"]  # Image resolution conversion factor (pixels to mm)
+    img_resample = params["resample"]  # Resampling factor for image resizing
+    bknd_img_path = params["background_img"]  # Path to the background image
+    E_max = params["Max_Eccentricity"]  # Maximum eccentricity threshold
+    S_min = params["Min_Solidity"]  # Minimum solidity threshold
+    Dmin = params["min_size"]  # Minimum bubble size (in mm)
```

Comment:
Give meaningful, human-readable names to the variables, so the code can be understood. Something like the following is longer but reads much better. 

```suggestion
    element_size = morphology.disk(
        params["Morphological_element_size"]
    )  # Structuring element for morphological operations
    connectivity = params["Connectivity"]  # Neighborhood connectivity (4 or 8)
    marker_size = params["Marker_size"]  # Marker size for watershed segmentation
    px2mm = params["px2mm"]  # Image resolution conversion factor (pixels to mm)
    img_resample = params["resample"]  # Resampling factor for image resizing
    bknd_img_path = params["background_img"]  # Path to the background image
    max_eccentricity = params["Max_Eccentricity"]  # Maximum eccentricity threshold
    min_solidity = params["Min_Solidity"]  # Minimum solidity threshold
    min_bubble_size = params["min_size"]  # Minimum bubble size (in mm)
```

These are not equations, where you need to keep the names very short - it is code and needs to be readable.